In [10]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns


In [42]:
#load train file
url = "updated_train.xlsx"
    
df = pd.read_excel(url)

df1= df[['Monthly rent payment', ' number of all rooms in the house', '=1 Overcrowding by rooms', '=1 has toilet in the household',
 '=1 if the household has refrigerator','Total males in the household','Total persons in the household','years of schooling', '=1 if predominant material on the outside wall is waste material',
 '=1 if no floor at the household']]
         
df2 = df[['=1 if the house has ceiling','=1 if no water provision','=1 no electricity in the dwelling',
 '=1 no main source of energy used for cooking (no kitchen)','=1 if disable person','=1 if male']]

df1.head()



Monthly rent payment   number of all rooms in the house  \
0              190000.0                                  3   
1              135000.0                                  4   
2                   NaN                                  8   
3              180000.0                                  5   
4              180000.0                                  5   

   =1 Overcrowding by rooms  =1 has toilet in the household  \
0                         0                               1   
1                         0                               1   
2                         0                               1   
3                         0                               1   
4                         0                               1   

   =1 if the household has refrigerator  Total males in the household  \
0                                     1                             1   
1                                     1                             1   
2                                     1                             0   
3                                     1                             2   
4                                     1                             2   

   Total persons in the household  years of schooling  \
0                               1                  10   
1                               1                  12   
2                               1                  11   
3                               4                   9   
4                               4                  11   

   =1 if predominant material on the outside wall is waste material  \
0                                                  0                  
1                                                  0                  
2                                                  0                  
3                                                  0                  
4                                                  0                  

   =1 if no floor at the household  
0                                0  
1                                0  
2                                0  
3                                0  
4                                0

In [21]:
list(df)

['Id',
 'Monthly rent payment',
 ' number of all rooms in the house',
 '=1 Overcrowding by rooms',
 '=1 has toilet in the household',
 '=1 if the household has refrigerator',
 'Total males in the household',
 'Total persons in the household',
 'years of schooling',
 '=1 if predominant material on the outside wall is waste material',
 '=1 if no floor at the household',
 '=1 if the house has ceiling',
 '=1 if no water provision',
 '=1 no electricity in the dwelling',
 '=1 no main source of energy used for cooking (no kitchen)',
 '=1 if disable person',
 '=1 if male',
 '=1 if divorced',
 '=1 if widow/er',
 '=1 if single',
 'Household level identifier',
 'Number of children 0 to 19 in household',
 'Dependency rate',
 'years of education of male head of household',
 'years of education of female head of household',
 '# persons per room',
 '=1 own and fully paid house',
 '=1 rented',
 '=1 region Central',
 '=1 region Chorotega',
 '=1 region PacÃÂÂ\xadfico central',
 '=1 region Brunca',
 

In [44]:
#Create a correlation matrix
corr = df.corr(numeric_only = True).round(3)
corr.style.background_gradient(cmap='coolwarm')

Monthly rent payment   number of all rooms in the house       0.443
                      =1 Overcrowding by rooms               -0.074
                      =1 has toilet in the household          0.034
                      =1 if the household has refrigerator    0.089
                      Total males in the household           -0.060
                                                              ...  
Target                =1 region Huetar Atlántica      -0.090
                      =1 region Huetar Norte                 -0.043
                      =1 zona urbana                          0.087
                      =2 zona rural                          -0.087
                      Age in years                            0.118
Length: 1042, dtype: float64

In [71]:
thresh = 0.7

#mask = corr.abs().lt(thresh)
mask = corr > thresh

df1 = corr.where(mask).stack()

df1.tail(20)


=1 if disable person                     =1 if disable person                       1.000
=1 if male                               =1 if male                                 1.000
=1 if divorced                           =1 if divorced                             1.000
=1 if widow/er                           =1 if widow/er                             1.000
=1 if single                             =1 if single                               1.000
Number of children 0 to 19 in household  Total persons in the household             0.751
                                         Number of children 0 to 19 in household    1.000
# persons per room                       # persons per room                         1.000
=1 own and fully paid house              =1 own and fully paid house                1.000
=1 rented                                =1 rented                                  1.000
=1 region Central                        =1 region Central                          1.000
=1 region 

In [75]:
thresh = 0.5

#mask = corr.abs().lt(thresh)
mask = corr >= thresh

df2 = corr.where(mask).stack()

df2.head(30)


TypeError: filter expected 2 arguments, got 1

In [79]:
thresh = 0.4

#mask = corr.abs().lt(thresh)
mask = corr >= thresh

df2 = corr.where(mask).stack()

df2.head(25)

Monthly rent payment                                              Monthly rent payment                                                1.000
                                                                   number of all rooms in the house                                   0.443
 number of all rooms in the house                                 Monthly rent payment                                                0.443
                                                                   number of all rooms in the house                                   1.000
=1 Overcrowding by rooms                                          =1 Overcrowding by rooms                                            1.000
                                                                  # persons per room                                                  0.530
=1 has toilet in the household                                    =1 has toilet in the household                                      1.000
=1 if the household 

# "Critical correlations" 

**Above 0.7**


| Variable 1 | Variable 2  | Correlation|
| --- | --- | --- |
| Total persons in the household | Total males in the household  | 0.735|
| Total persons in the household | Number of children 0 to 19 in household |   0.751|


**Above 0.5**

| Variable 1 | Variable 2  | Correlation|
| --- | --- | --- |
| #persons per room   | =1 Overcrowding by rooms       |            0.530|
| ---|   Total persons in the household         |    0.583|
| --- | Number of children 0 to 19 in household  |  0.596|
                                         
                                         
**Above 0.4**

| Variable 1 | Variable 2  | Correlation|
| --- | --- | --- |
|#persons per room  |                    Total males in the household|             0.406|
| Number of children 0 to 19 in household |Total males in the household   |          0.467|
|Monthly rent payment              |     number of all rooms in the house  |        0.443|
| 1 if predominant material on the outside wall is waste material   |=1 if no floor at the household   |        0.434|